server/src/main/java/com/scalar/db/server/DistributedStorageService.java

```diff
@@ -206,8 +207,10 @@ public void onError(Throwable t) {
 
     @Override
     public void onCompleted() {
-      responseObserver.onCompleted();
-      cleanUp();
+      if (!cleanedUp.get()) {
```

Comment:
I'm not sure but, is there any case where `cleanedUp` is set to true but `onCompleted` is not called?
If there is such a case, `onCompleted` won't be called properly?

Reply:
The server side proactively cleans up only when the scanner has no more results. So when a user calls `closeScanner` before the server cleans up, the case will happen.